packages/integration/config.d.ts

```diff
@@ -163,5 +163,26 @@ export interface Config {
        */
       privateKey?: string;
     };
+
+    /** Integration configuration for AWS S3 Service */
+    awsS3?: Array<{
+      /**
+       * The host of the target that this matches on, e.g. "amazonaws.com".
+       *
+       * @visibility frontend
+       */
+      host: string;
+      /**
+      /**
+       * Account access key used to authenticate requests.
+       * @visibility backend
+       */
+      accessKeyID?: string;
```

Comment:
I'm wondering if we need to make this a little more abstract here to support different forms of authentication with AWS? What about roles? would they be included here too?

Reply:
Think this should also be `accessKeyId`

Reply:
Hello, it looks like something similar is being done in the [Techdocs Backend](https://github.com/backstage/backstage/blob/master/plugins/techdocs-backend/config.d.ts) where `roleArn` is included alongside `accessKeyId` and `secretAccessKey`. Is this the direction we should consider for our use case?

Reply:
I think it could be a good approach to follow the logic within `AwsS3Publish.buildCredentials` to generate the credentials object. That uses these two access & secret keys if they exist, defaults to using standard credentials provider chain if they dont, and finally optionally wraps the a role assumption around those credentials is roleArn is defined. 

For AWS stuff it might be useful to generalize this pattern IMO.

Reply:
We followed the logic for `AwsS3Publish.buildCredentials` and incorporated `roleArn` in our configurations.